### ISSUE TITLE
Security Audit

### ISSUE THREAD

**Issue by johnSchnake** [heptio/sonobuoy#752]
**Describe the solution you'd like**
Are there fields we want to prevent users from setting on the pods? Privileged containers? hostPID?

There are a few features which may be implemented by making more fields like this customizable by the user and I don't know if there is a line in the sand we need to draw so as not to open up too much attack surface.

A similar concern for implementing an API server in the aggregator to enable users to do things like delete/check status/etc etc. Adding an API layer and functionality during the run may add extra attack surface.

**Comment by johnSchnake** [comment]
Completed.

TL;DR; 
 - No concerns for setting certain fields on pods/etc
 - Daemonsets currently seem to set some pod settings that are not necessary, violating the principle of least access
   - Since we can make any fields editable, we can eventually resolve that. I think the daemonsets have some of those permissions in order to get the systemd-logs plugin to work so we can just move the settings from daemonset wide to plugin specific.
 - The API on the aggregator is a can of worms to ensure we don't open the users up to privilege escalation. There may be some other way to achieve our same goal (in general, we'd like a few buttons the user can wack on the running pod to get some behavior to occur) but based on the number of concerns making a secure API (where no API currently exists) the ROI doesn't seem favorable.
